sdk/typescript/src/api/client.ts

```diff
@@ -0,0 +1,814 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+import type {
+	ExecuteTransactionRequestType,
+	ObjectId,
+	SuiEventFilter,
+	TransactionDigest,
+	SuiTransactionBlockResponseQuery,
+	Order,
+	CoinMetadata,
+	CheckpointDigest,
+	SuiObjectDataOptions,
+	SuiTransactionBlockResponseOptions,
+	SuiEvent,
+	SuiObjectResponseQuery,
+	TransactionFilter,
+	TransactionEffects,
+	Unsubscribe,
+	PaginatedTransactionResponse,
+	SuiAddress,
+	SuiMoveFunctionArgTypes,
+	SuiMoveNormalizedFunction,
+	SuiMoveNormalizedModule,
+	SuiMoveNormalizedModules,
+	SuiMoveNormalizedStruct,
+	SuiTransactionBlockResponse,
+	PaginatedEvents,
+	DevInspectResults,
+	PaginatedCoins,
+	SuiObjectResponse,
+	DelegatedStake,
+	CoinBalance,
+	CoinSupply,
+	Checkpoint,
+	CommitteeInfo,
+	DryRunTransactionBlockResponse,
+	SuiSystemStateSummary,
+	PaginatedObjectsResponse,
+	ValidatorsApy,
+	MoveCallMetrics,
+	ObjectRead,
+	ResolvedNameServiceNames,
+	ProtocolConfig,
+	EpochInfo,
+	EpochPage,
+	CheckpointPage,
+	DynamicFieldName,
+	DynamicFieldPage,
+	NetworkMetrics,
+	AddressMetrics,
+} from '../types/index.js';
+import {
+	isValidTransactionDigest,
+	isValidSuiAddress,
+	isValidSuiObjectId,
+	normalizeSuiAddress,
+	normalizeSuiObjectId,
+} from '../utils/sui-types.js';
+import { fromB58, toB64, toHEX } from '@mysten/bcs';
+import type { SerializedSignature } from '../cryptography/signature.js';
+import { TransactionBlock } from '../builder/index.js';
+import { SuiHTTPTransport } from './http-transport.js';
+import type { SuiNetwork, SuiTransport } from './http-transport.js';
+
+export interface PaginationArguments<Cursor> {
+	/** Optional paging cursor */
+	cursor?: Cursor;
+	/** Maximum item returned per page */
+	limit?: number | null;
+}
+
+export interface OrderArguments {
+	order?: Order | null;
+}
+
+/**
+ * Configuration options for the JsonRpcProvider. If the value of a field is not provided,
+ * value in `DEFAULT_OPTIONS` for that field will be used
+ */
+export type SuiApiClientOptions = NetworkOrTransport & {
+	/**
+	 * Cache timeout in seconds for the RPC API Version
+	 */
+	versionCacheTimeoutInSeconds?: number;
+};
+
+export type NetworkOrTransport =
+	| {
+			network: SuiNetwork;
+			transport?: never;
+	  }
+	| {
+			transport: SuiTransport;
+			network?: never;
+	  };
+
+const DEFAULT_OPTIONS = {
+	versionCacheTimeoutInSeconds: 600,
+} satisfies Partial<SuiApiClientOptions>;
+
+export class SuiAPIClient {
```

Comment:
I don't love this name. I don't know why you need to disambiguate with `Sui`. I honestly think `Client` might be fine?

Reply:
Happy to defer to you on this, but I generally like more specific names in libraries because you often end up with More than one client in an App.  It makes auto-imports better, and avoids having to alias things in your app. How do you feel about `SuiClient`?